dll/win32/shell32/CDefaultContextMenu.cpp

```diff
@@ -1260,6 +1260,13 @@ CDefaultContextMenu::BrowserFlagsFromVerb(LPCMINVOKECOMMANDINFOEX lpcmi, PStatic
     else
         FlagsName = L"BrowserFlags";
 
+    CComPtr<ICommDlgBrowser> pcdb;
+    if (SUCCEEDED(psb->QueryInterface(IID_PPV_ARG(ICommDlgBrowser, &pcdb))))
+    {
+        if (LOBYTE(GetVersion()) < 6 || FlagsName[0] == 'E')
+            return 0; // Don't browse in-place
```

Comment:
Can you quickly explain to me what this "Explore" vs. "Browse" mode is?

Reply:
Open vs Explore verbs in the context menu are supposed to handle their commands differently depending on if the folder tree is present in Explorer. For example, if the tree is not present and you right-click a folder and select Explore, it's supposed to open a new Explorer window as with `explore.exe /e,c:\foo`. These registry values are supposed to let the verb override this (I don't know why). The behavior is different between NT5 and 6 and ROS is somewhere in between IIRC.

In this case, we are in the Open/Save dialog and we always want a new window for Explore.